# Practice C. Increase the number of epochs - Practice B but with 100 epochs
- Repeat part A but use a normalized version of the data & epcohs = 100
#####  Dataset
- [Download dataset here](https://cocl.us/concrete_data)

In [1]:
# Import packages
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

import keras
from keras.models import Sequential
from keras.layers import Dense

from sklearn.metrics import mean_squared_error
from sklearn.model_selection import train_test_split

# Read in the data
concrete_data = pd.read_csv("/Users/pc/Desktop/IBM AI Engineer/Introduction to Deep Learning & Neural Networks with Keras/concrete_data.csv")

# Split data into predictors and target
concrete_data_columns = concrete_data.columns
predictors = concrete_data[concrete_data_columns[concrete_data_columns != "Strength"]]
target = concrete_data["Strength"]

2023-08-16 15:09:11.015658: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.


In [2]:
# Normalization of the data
predictors_norm = (predictors - predictors.mean()) / predictors.std()
target = (target - target.mean()) / target.std()

# the number of predictors
n_cols = predictors_norm.shape[1]

In [3]:
# define the regression model
def regression_model():
    # create model
    model = Sequential()
    model.add(Dense(10, activation = "relu", input_shape = (n_cols, )))
    model.add(Dense(1))
    
    # compile model
    model.compile(optimizer = "adam", loss = "mean_squared_error")
    return model

# build the model
model = regression_model()

### Epochs = 100

In [4]:
MSE_list = []
trial_times = 50
for t in range(trial_times):
    # train / test split
    X_train, X_test, y_train, y_test = train_test_split(predictors, 
                                                    target, 
                                                    test_size=0.3, 
                                                    random_state=42)
    # fit the model
    model.fit(X_train, y_train, epochs = 100, verbose = 2)
    
    # predict values
    predictions = model.predict(X_test)

    # Mean Squared Errors between predictions and the actual values: y_test
    predictions = predictions.reshape(predictions.size, )
    MSE_errors = mean_squared_error(y_test, predictions)
    MSE_list.append(MSE_errors)
    #print("The MSE errors at the {} epoch is {}.".format(e + 1, MSE_errors))


Epoch 1/100
23/23 - 0s - loss: 7398.6504 - 477ms/epoch - 21ms/step
Epoch 2/100
23/23 - 0s - loss: 4566.6191 - 20ms/epoch - 873us/step
Epoch 3/100
23/23 - 0s - loss: 3057.3416 - 22ms/epoch - 936us/step
Epoch 4/100
23/23 - 0s - loss: 2006.3019 - 22ms/epoch - 937us/step
Epoch 5/100
23/23 - 0s - loss: 1407.9225 - 20ms/epoch - 876us/step
Epoch 6/100
23/23 - 0s - loss: 1087.4811 - 21ms/epoch - 898us/step
Epoch 7/100
23/23 - 0s - loss: 834.6879 - 26ms/epoch - 1ms/step
Epoch 8/100
23/23 - 0s - loss: 662.3228 - 23ms/epoch - 1ms/step
Epoch 9/100
23/23 - 0s - loss: 522.1813 - 23ms/epoch - 989us/step
Epoch 10/100
23/23 - 0s - loss: 416.2350 - 19ms/epoch - 835us/step
Epoch 11/100
23/23 - 0s - loss: 327.8706 - 20ms/epoch - 850us/step
Epoch 12/100
23/23 - 0s - loss: 259.8545 - 27ms/epoch - 1ms/step
Epoch 13/100
23/23 - 0s - loss: 203.8829 - 21ms/epoch - 922us/step
Epoch 14/100
23/23 - 0s - loss: 164.3190 - 34ms/epoch - 1ms/step
Epoch 15/100
23/23 - 0s - loss: 129.4816 - 27ms/epoch - 1ms/step
Epoch 16

Epoch 27/100
23/23 - 0s - loss: 0.4614 - 17ms/epoch - 730us/step
Epoch 28/100
23/23 - 0s - loss: 0.4661 - 16ms/epoch - 705us/step
Epoch 29/100
23/23 - 0s - loss: 0.4517 - 21ms/epoch - 924us/step
Epoch 30/100
23/23 - 0s - loss: 0.4910 - 21ms/epoch - 925us/step
Epoch 31/100
23/23 - 0s - loss: 0.4695 - 17ms/epoch - 726us/step
Epoch 32/100
23/23 - 0s - loss: 0.4829 - 16ms/epoch - 684us/step
Epoch 33/100
23/23 - 0s - loss: 0.4493 - 16ms/epoch - 683us/step
Epoch 34/100
23/23 - 0s - loss: 0.4744 - 17ms/epoch - 719us/step
Epoch 35/100
23/23 - 0s - loss: 0.4252 - 16ms/epoch - 708us/step
Epoch 36/100
23/23 - 0s - loss: 0.4281 - 19ms/epoch - 837us/step
Epoch 37/100
23/23 - 0s - loss: 0.4589 - 16ms/epoch - 689us/step
Epoch 38/100
23/23 - 0s - loss: 0.4404 - 16ms/epoch - 685us/step
Epoch 39/100
23/23 - 0s - loss: 0.4440 - 16ms/epoch - 700us/step
Epoch 40/100
23/23 - 0s - loss: 0.4321 - 19ms/epoch - 816us/step
Epoch 41/100
23/23 - 0s - loss: 0.4583 - 16ms/epoch - 701us/step
Epoch 42/100
23/23 - 0s -

Epoch 53/100
23/23 - 0s - loss: 0.4658 - 18ms/epoch - 789us/step
Epoch 54/100
23/23 - 0s - loss: 0.5009 - 19ms/epoch - 822us/step
Epoch 55/100
23/23 - 0s - loss: 0.5378 - 19ms/epoch - 829us/step
Epoch 56/100
23/23 - 0s - loss: 0.8381 - 18ms/epoch - 786us/step
Epoch 57/100
23/23 - 0s - loss: 0.5395 - 18ms/epoch - 800us/step
Epoch 58/100
23/23 - 0s - loss: 0.5853 - 19ms/epoch - 817us/step
Epoch 59/100
23/23 - 0s - loss: 0.5980 - 18ms/epoch - 784us/step
Epoch 60/100
23/23 - 0s - loss: 0.4393 - 26ms/epoch - 1ms/step
Epoch 61/100
23/23 - 0s - loss: 0.6116 - 32ms/epoch - 1ms/step
Epoch 62/100
23/23 - 0s - loss: 0.6713 - 35ms/epoch - 2ms/step
Epoch 63/100
23/23 - 0s - loss: 0.5176 - 26ms/epoch - 1ms/step
Epoch 64/100
23/23 - 0s - loss: 0.4641 - 23ms/epoch - 1ms/step
Epoch 65/100
23/23 - 0s - loss: 0.5336 - 19ms/epoch - 830us/step
Epoch 66/100
23/23 - 0s - loss: 0.5930 - 20ms/epoch - 849us/step
Epoch 67/100
23/23 - 0s - loss: 0.6013 - 19ms/epoch - 846us/step
Epoch 68/100
23/23 - 0s - loss: 0.7

Epoch 79/100
23/23 - 0s - loss: 0.4728 - 17ms/epoch - 761us/step
Epoch 80/100
23/23 - 0s - loss: 0.6067 - 19ms/epoch - 829us/step
Epoch 81/100
23/23 - 0s - loss: 0.4393 - 18ms/epoch - 795us/step
Epoch 82/100
23/23 - 0s - loss: 0.6361 - 16ms/epoch - 709us/step
Epoch 83/100
23/23 - 0s - loss: 0.5439 - 16ms/epoch - 697us/step
Epoch 84/100
23/23 - 0s - loss: 1.0482 - 16ms/epoch - 699us/step
Epoch 85/100
23/23 - 0s - loss: 0.5782 - 35ms/epoch - 2ms/step
Epoch 86/100
23/23 - 0s - loss: 0.4383 - 24ms/epoch - 1ms/step
Epoch 87/100
23/23 - 0s - loss: 0.5149 - 20ms/epoch - 855us/step
Epoch 88/100
23/23 - 0s - loss: 0.5497 - 19ms/epoch - 806us/step
Epoch 89/100
23/23 - 0s - loss: 0.4525 - 17ms/epoch - 729us/step
Epoch 90/100
23/23 - 0s - loss: 0.5462 - 16ms/epoch - 712us/step
Epoch 91/100
23/23 - 0s - loss: 0.4985 - 16ms/epoch - 706us/step
Epoch 92/100
23/23 - 0s - loss: 0.4731 - 17ms/epoch - 718us/step
Epoch 93/100
23/23 - 0s - loss: 0.5795 - 17ms/epoch - 724us/step
Epoch 94/100
23/23 - 0s - los

Epoch 4/100
23/23 - 0s - loss: 0.7220 - 16ms/epoch - 678us/step
Epoch 5/100
23/23 - 0s - loss: 0.5800 - 16ms/epoch - 686us/step
Epoch 6/100
23/23 - 0s - loss: 0.5321 - 16ms/epoch - 691us/step
Epoch 7/100
23/23 - 0s - loss: 0.4637 - 15ms/epoch - 672us/step
Epoch 8/100
23/23 - 0s - loss: 0.4824 - 16ms/epoch - 685us/step
Epoch 9/100
23/23 - 0s - loss: 0.9979 - 18ms/epoch - 777us/step
Epoch 10/100
23/23 - 0s - loss: 0.9161 - 16ms/epoch - 685us/step
Epoch 11/100
23/23 - 0s - loss: 0.5533 - 15ms/epoch - 659us/step
Epoch 12/100
23/23 - 0s - loss: 0.4827 - 17ms/epoch - 727us/step
Epoch 13/100
23/23 - 0s - loss: 0.4322 - 16ms/epoch - 709us/step
Epoch 14/100
23/23 - 0s - loss: 0.6270 - 16ms/epoch - 694us/step
Epoch 15/100
23/23 - 0s - loss: 0.4997 - 23ms/epoch - 994us/step
Epoch 16/100
23/23 - 0s - loss: 0.7122 - 17ms/epoch - 727us/step
Epoch 17/100
23/23 - 0s - loss: 0.5926 - 16ms/epoch - 697us/step
Epoch 18/100
23/23 - 0s - loss: 0.5349 - 16ms/epoch - 675us/step
Epoch 19/100
23/23 - 0s - loss:

Epoch 30/100
23/23 - 0s - loss: 0.6031 - 17ms/epoch - 739us/step
Epoch 31/100
23/23 - 0s - loss: 0.6060 - 17ms/epoch - 736us/step
Epoch 32/100
23/23 - 0s - loss: 0.4996 - 17ms/epoch - 727us/step
Epoch 33/100
23/23 - 0s - loss: 0.4618 - 18ms/epoch - 784us/step
Epoch 34/100
23/23 - 0s - loss: 0.6607 - 18ms/epoch - 804us/step
Epoch 35/100
23/23 - 0s - loss: 0.6322 - 17ms/epoch - 759us/step
Epoch 36/100
23/23 - 0s - loss: 0.5266 - 17ms/epoch - 737us/step
Epoch 37/100
23/23 - 0s - loss: 0.5495 - 17ms/epoch - 719us/step
Epoch 38/100
23/23 - 0s - loss: 0.4990 - 17ms/epoch - 760us/step
Epoch 39/100
23/23 - 0s - loss: 0.4712 - 17ms/epoch - 728us/step
Epoch 40/100
23/23 - 0s - loss: 0.4769 - 17ms/epoch - 749us/step
Epoch 41/100
23/23 - 0s - loss: 0.4617 - 17ms/epoch - 760us/step
Epoch 42/100
23/23 - 0s - loss: 0.4802 - 17ms/epoch - 738us/step
Epoch 43/100
23/23 - 0s - loss: 0.4416 - 58ms/epoch - 3ms/step
Epoch 44/100
23/23 - 0s - loss: 0.4662 - 17ms/epoch - 731us/step
Epoch 45/100
23/23 - 0s - l

Epoch 56/100
23/23 - 0s - loss: 0.6074 - 18ms/epoch - 789us/step
Epoch 57/100
23/23 - 0s - loss: 0.5157 - 18ms/epoch - 780us/step
Epoch 58/100
23/23 - 0s - loss: 0.4630 - 19ms/epoch - 814us/step
Epoch 59/100
23/23 - 0s - loss: 0.4927 - 17ms/epoch - 756us/step
Epoch 60/100
23/23 - 0s - loss: 0.4517 - 17ms/epoch - 749us/step
Epoch 61/100
23/23 - 0s - loss: 0.5190 - 17ms/epoch - 718us/step
Epoch 62/100
23/23 - 0s - loss: 0.5425 - 17ms/epoch - 731us/step
Epoch 63/100
23/23 - 0s - loss: 0.5798 - 16ms/epoch - 712us/step
Epoch 64/100
23/23 - 0s - loss: 0.4792 - 16ms/epoch - 715us/step
Epoch 65/100
23/23 - 0s - loss: 0.7109 - 17ms/epoch - 718us/step
Epoch 66/100
23/23 - 0s - loss: 0.7785 - 16ms/epoch - 707us/step
Epoch 67/100
23/23 - 0s - loss: 0.7591 - 16ms/epoch - 712us/step
Epoch 68/100
23/23 - 0s - loss: 0.7892 - 16ms/epoch - 706us/step
Epoch 69/100
23/23 - 0s - loss: 0.4874 - 19ms/epoch - 841us/step
Epoch 70/100
23/23 - 0s - loss: 0.6404 - 17ms/epoch - 750us/step
Epoch 71/100
23/23 - 0s -

Epoch 82/100
23/23 - 0s - loss: 0.5712 - 19ms/epoch - 809us/step
Epoch 83/100
23/23 - 0s - loss: 0.4875 - 18ms/epoch - 778us/step
Epoch 84/100
23/23 - 0s - loss: 0.5334 - 20ms/epoch - 858us/step
Epoch 85/100
23/23 - 0s - loss: 0.4910 - 20ms/epoch - 888us/step
Epoch 86/100
23/23 - 0s - loss: 0.5370 - 18ms/epoch - 788us/step
Epoch 87/100
23/23 - 0s - loss: 0.5920 - 18ms/epoch - 775us/step
Epoch 88/100
23/23 - 0s - loss: 0.4890 - 23ms/epoch - 1ms/step
Epoch 89/100
23/23 - 0s - loss: 0.5322 - 20ms/epoch - 851us/step
Epoch 90/100
23/23 - 0s - loss: 0.5550 - 19ms/epoch - 813us/step
Epoch 91/100
23/23 - 0s - loss: 0.4769 - 23ms/epoch - 1ms/step
Epoch 92/100
23/23 - 0s - loss: 0.4658 - 26ms/epoch - 1ms/step
Epoch 93/100
23/23 - 0s - loss: 1.0363 - 26ms/epoch - 1ms/step
Epoch 94/100
23/23 - 0s - loss: 0.5433 - 23ms/epoch - 1ms/step
Epoch 95/100
23/23 - 0s - loss: 0.4937 - 22ms/epoch - 977us/step
Epoch 96/100
23/23 - 0s - loss: 0.4539 - 20ms/epoch - 864us/step
Epoch 97/100
23/23 - 0s - loss: 0.4

Epoch 7/100
23/23 - 0s - loss: 0.5011 - 17ms/epoch - 731us/step
Epoch 8/100
23/23 - 0s - loss: 0.5329 - 16ms/epoch - 693us/step
Epoch 9/100
23/23 - 0s - loss: 0.5799 - 16ms/epoch - 690us/step
Epoch 10/100
23/23 - 0s - loss: 0.5104 - 16ms/epoch - 685us/step
Epoch 11/100
23/23 - 0s - loss: 0.6565 - 16ms/epoch - 679us/step
Epoch 12/100
23/23 - 0s - loss: 0.5883 - 16ms/epoch - 685us/step
Epoch 13/100
23/23 - 0s - loss: 0.4437 - 16ms/epoch - 711us/step
Epoch 14/100
23/23 - 0s - loss: 0.6367 - 16ms/epoch - 685us/step
Epoch 15/100
23/23 - 0s - loss: 0.4924 - 16ms/epoch - 676us/step
Epoch 16/100
23/23 - 0s - loss: 0.4388 - 16ms/epoch - 693us/step
Epoch 17/100
23/23 - 0s - loss: 0.4930 - 16ms/epoch - 699us/step
Epoch 18/100
23/23 - 0s - loss: 0.4795 - 16ms/epoch - 697us/step
Epoch 19/100
23/23 - 0s - loss: 0.5752 - 17ms/epoch - 725us/step
Epoch 20/100
23/23 - 0s - loss: 0.5055 - 16ms/epoch - 686us/step
Epoch 21/100
23/23 - 0s - loss: 0.4972 - 17ms/epoch - 753us/step
Epoch 22/100
23/23 - 0s - lo

Epoch 33/100
23/23 - 0s - loss: 0.4879 - 18ms/epoch - 764us/step
Epoch 34/100
23/23 - 0s - loss: 0.4794 - 17ms/epoch - 729us/step
Epoch 35/100
23/23 - 0s - loss: 0.5570 - 17ms/epoch - 718us/step
Epoch 36/100
23/23 - 0s - loss: 0.4406 - 16ms/epoch - 681us/step
Epoch 37/100
23/23 - 0s - loss: 0.6297 - 16ms/epoch - 679us/step
Epoch 38/100
23/23 - 0s - loss: 0.5040 - 16ms/epoch - 693us/step
Epoch 39/100
23/23 - 0s - loss: 0.6837 - 16ms/epoch - 697us/step
Epoch 40/100
23/23 - 0s - loss: 0.5662 - 16ms/epoch - 688us/step
Epoch 41/100
23/23 - 0s - loss: 0.5068 - 16ms/epoch - 676us/step
Epoch 42/100
23/23 - 0s - loss: 0.4186 - 16ms/epoch - 676us/step
Epoch 43/100
23/23 - 0s - loss: 0.4531 - 15ms/epoch - 656us/step
Epoch 44/100
23/23 - 0s - loss: 0.4724 - 15ms/epoch - 666us/step
Epoch 45/100
23/23 - 0s - loss: 0.4854 - 15ms/epoch - 672us/step
Epoch 46/100
23/23 - 0s - loss: 0.5041 - 15ms/epoch - 669us/step
Epoch 47/100
23/23 - 0s - loss: 0.5704 - 15ms/epoch - 661us/step
Epoch 48/100
23/23 - 0s -

Epoch 59/100
23/23 - 0s - loss: 0.4680 - 16ms/epoch - 678us/step
Epoch 60/100
23/23 - 0s - loss: 0.5210 - 16ms/epoch - 699us/step
Epoch 61/100
23/23 - 0s - loss: 0.5125 - 16ms/epoch - 715us/step
Epoch 62/100
23/23 - 0s - loss: 0.4854 - 16ms/epoch - 704us/step
Epoch 63/100
23/23 - 0s - loss: 0.5396 - 16ms/epoch - 681us/step
Epoch 64/100
23/23 - 0s - loss: 0.7504 - 18ms/epoch - 786us/step
Epoch 65/100
23/23 - 0s - loss: 0.4964 - 17ms/epoch - 747us/step
Epoch 66/100
23/23 - 0s - loss: 0.4258 - 18ms/epoch - 772us/step
Epoch 67/100
23/23 - 0s - loss: 0.5829 - 17ms/epoch - 720us/step
Epoch 68/100
23/23 - 0s - loss: 0.6849 - 17ms/epoch - 759us/step
Epoch 69/100
23/23 - 0s - loss: 0.4814 - 18ms/epoch - 773us/step
Epoch 70/100
23/23 - 0s - loss: 0.4520 - 18ms/epoch - 764us/step
Epoch 71/100
23/23 - 0s - loss: 0.5147 - 17ms/epoch - 732us/step
Epoch 72/100
23/23 - 0s - loss: 0.4852 - 17ms/epoch - 732us/step
Epoch 73/100
23/23 - 0s - loss: 0.6054 - 16ms/epoch - 708us/step
Epoch 74/100
23/23 - 0s -

Epoch 85/100
23/23 - 0s - loss: 0.4511 - 16ms/epoch - 703us/step
Epoch 86/100
23/23 - 0s - loss: 0.4692 - 28ms/epoch - 1ms/step
Epoch 87/100
23/23 - 0s - loss: 0.5195 - 30ms/epoch - 1ms/step
Epoch 88/100
23/23 - 0s - loss: 0.4958 - 20ms/epoch - 848us/step
Epoch 89/100
23/23 - 0s - loss: 0.5172 - 19ms/epoch - 830us/step
Epoch 90/100
23/23 - 0s - loss: 0.6610 - 21ms/epoch - 907us/step
Epoch 91/100
23/23 - 0s - loss: 0.5259 - 19ms/epoch - 830us/step
Epoch 92/100
23/23 - 0s - loss: 0.4588 - 17ms/epoch - 744us/step
Epoch 93/100
23/23 - 0s - loss: 0.6342 - 18ms/epoch - 795us/step
Epoch 94/100
23/23 - 0s - loss: 1.0251 - 18ms/epoch - 767us/step
Epoch 95/100
23/23 - 0s - loss: 0.7576 - 18ms/epoch - 763us/step
Epoch 96/100
23/23 - 0s - loss: 0.4626 - 18ms/epoch - 763us/step
Epoch 97/100
23/23 - 0s - loss: 0.4513 - 18ms/epoch - 766us/step
Epoch 98/100
23/23 - 0s - loss: 0.4973 - 18ms/epoch - 772us/step
Epoch 99/100
23/23 - 0s - loss: 0.5126 - 18ms/epoch - 779us/step
Epoch 100/100
23/23 - 0s - lo

23/23 - 0s - loss: 0.5631 - 19ms/epoch - 824us/step
Epoch 11/100
23/23 - 0s - loss: 0.5281 - 19ms/epoch - 821us/step
Epoch 12/100
23/23 - 0s - loss: 0.4731 - 18ms/epoch - 773us/step
Epoch 13/100
23/23 - 0s - loss: 0.5586 - 18ms/epoch - 783us/step
Epoch 14/100
23/23 - 0s - loss: 0.4540 - 19ms/epoch - 825us/step
Epoch 15/100
23/23 - 0s - loss: 0.5035 - 17ms/epoch - 727us/step
Epoch 16/100
23/23 - 0s - loss: 0.4432 - 18ms/epoch - 803us/step
Epoch 17/100
23/23 - 0s - loss: 0.7230 - 18ms/epoch - 787us/step
Epoch 18/100
23/23 - 0s - loss: 0.6906 - 18ms/epoch - 794us/step
Epoch 19/100
23/23 - 0s - loss: 0.6925 - 18ms/epoch - 776us/step
Epoch 20/100
23/23 - 0s - loss: 0.5293 - 18ms/epoch - 773us/step
Epoch 21/100
23/23 - 0s - loss: 0.8842 - 35ms/epoch - 2ms/step
Epoch 22/100
23/23 - 0s - loss: 0.7848 - 30ms/epoch - 1ms/step
Epoch 23/100
23/23 - 0s - loss: 1.1650 - 23ms/epoch - 992us/step
Epoch 24/100
23/23 - 0s - loss: 0.7264 - 21ms/epoch - 911us/step
Epoch 25/100
23/23 - 0s - loss: 0.5734 - 2

Epoch 37/100
23/23 - 0s - loss: 0.4387 - 19ms/epoch - 824us/step
Epoch 38/100
23/23 - 0s - loss: 0.4575 - 23ms/epoch - 1ms/step
Epoch 39/100
23/23 - 0s - loss: 0.4898 - 19ms/epoch - 828us/step
Epoch 40/100
23/23 - 0s - loss: 0.4380 - 18ms/epoch - 783us/step
Epoch 41/100
23/23 - 0s - loss: 0.5696 - 18ms/epoch - 778us/step
Epoch 42/100
23/23 - 0s - loss: 0.4963 - 18ms/epoch - 796us/step
Epoch 43/100
23/23 - 0s - loss: 0.5412 - 20ms/epoch - 851us/step
Epoch 44/100
23/23 - 0s - loss: 0.5817 - 18ms/epoch - 783us/step
Epoch 45/100
23/23 - 0s - loss: 0.4675 - 18ms/epoch - 799us/step
Epoch 46/100
23/23 - 0s - loss: 0.7949 - 21ms/epoch - 924us/step
Epoch 47/100
23/23 - 0s - loss: 0.5302 - 20ms/epoch - 870us/step
Epoch 48/100
23/23 - 0s - loss: 0.5711 - 19ms/epoch - 825us/step
Epoch 49/100
23/23 - 0s - loss: 0.5737 - 20ms/epoch - 856us/step
Epoch 50/100
23/23 - 0s - loss: 0.4559 - 19ms/epoch - 817us/step
Epoch 51/100
23/23 - 0s - loss: 0.5094 - 18ms/epoch - 797us/step
Epoch 52/100
23/23 - 0s - l

Epoch 63/100
23/23 - 0s - loss: 0.4763 - 16ms/epoch - 714us/step
Epoch 64/100
23/23 - 0s - loss: 0.5789 - 16ms/epoch - 705us/step
Epoch 65/100
23/23 - 0s - loss: 0.7843 - 16ms/epoch - 717us/step
Epoch 66/100
23/23 - 0s - loss: 0.4526 - 20ms/epoch - 862us/step
Epoch 67/100
23/23 - 0s - loss: 0.6731 - 16ms/epoch - 681us/step
Epoch 68/100
23/23 - 0s - loss: 0.5156 - 16ms/epoch - 679us/step
Epoch 69/100
23/23 - 0s - loss: 0.7921 - 15ms/epoch - 672us/step
Epoch 70/100
23/23 - 0s - loss: 0.4659 - 15ms/epoch - 667us/step
Epoch 71/100
23/23 - 0s - loss: 0.5454 - 16ms/epoch - 683us/step
Epoch 72/100
23/23 - 0s - loss: 0.4430 - 15ms/epoch - 668us/step
Epoch 73/100
23/23 - 0s - loss: 0.5155 - 15ms/epoch - 672us/step
Epoch 74/100
23/23 - 0s - loss: 0.5758 - 16ms/epoch - 713us/step
Epoch 75/100
23/23 - 0s - loss: 0.5562 - 16ms/epoch - 686us/step
Epoch 76/100
23/23 - 0s - loss: 0.5595 - 15ms/epoch - 671us/step
Epoch 77/100
23/23 - 0s - loss: 0.5517 - 16ms/epoch - 674us/step
Epoch 78/100
23/23 - 0s -

Epoch 89/100
23/23 - 0s - loss: 0.5650 - 17ms/epoch - 730us/step
Epoch 90/100
23/23 - 0s - loss: 0.5323 - 17ms/epoch - 737us/step
Epoch 91/100
23/23 - 0s - loss: 0.7793 - 17ms/epoch - 753us/step
Epoch 92/100
23/23 - 0s - loss: 0.5028 - 17ms/epoch - 741us/step
Epoch 93/100
23/23 - 0s - loss: 0.5383 - 17ms/epoch - 740us/step
Epoch 94/100
23/23 - 0s - loss: 0.5251 - 16ms/epoch - 705us/step
Epoch 95/100
23/23 - 0s - loss: 0.6537 - 16ms/epoch - 677us/step
Epoch 96/100
23/23 - 0s - loss: 0.6725 - 17ms/epoch - 721us/step
Epoch 97/100
23/23 - 0s - loss: 0.5021 - 16ms/epoch - 675us/step
Epoch 98/100
23/23 - 0s - loss: 0.4631 - 17ms/epoch - 746us/step
Epoch 99/100
23/23 - 0s - loss: 0.6656 - 18ms/epoch - 800us/step
Epoch 100/100
23/23 - 0s - loss: 0.5973 - 18ms/epoch - 767us/step
10/10 [==============================] - 0s 867us/step
Epoch 1/100
23/23 - 0s - loss: 0.4585 - 19ms/epoch - 825us/step
Epoch 2/100
23/23 - 0s - loss: 0.5025 - 17ms/epoch - 738us/step
Epoch 3/100
23/23 - 0s - loss: 0.456

Epoch 14/100
23/23 - 0s - loss: 0.5925 - 16ms/epoch - 689us/step
Epoch 15/100
23/23 - 0s - loss: 0.5257 - 15ms/epoch - 669us/step
Epoch 16/100
23/23 - 0s - loss: 0.5640 - 16ms/epoch - 684us/step
Epoch 17/100
23/23 - 0s - loss: 0.5122 - 21ms/epoch - 934us/step
Epoch 18/100
23/23 - 0s - loss: 0.6119 - 16ms/epoch - 708us/step
Epoch 19/100
23/23 - 0s - loss: 0.7648 - 15ms/epoch - 669us/step
Epoch 20/100
23/23 - 0s - loss: 0.5186 - 16ms/epoch - 680us/step
Epoch 21/100
23/23 - 0s - loss: 0.5149 - 17ms/epoch - 733us/step
Epoch 22/100
23/23 - 0s - loss: 0.6671 - 19ms/epoch - 813us/step
Epoch 23/100
23/23 - 0s - loss: 0.5517 - 20ms/epoch - 854us/step
Epoch 24/100
23/23 - 0s - loss: 0.5809 - 16ms/epoch - 680us/step
Epoch 25/100
23/23 - 0s - loss: 0.5052 - 16ms/epoch - 704us/step
Epoch 26/100
23/23 - 0s - loss: 0.4610 - 15ms/epoch - 657us/step
Epoch 27/100
23/23 - 0s - loss: 0.5127 - 16ms/epoch - 716us/step
Epoch 28/100
23/23 - 0s - loss: 0.5911 - 15ms/epoch - 671us/step
Epoch 29/100
23/23 - 0s -

Epoch 40/100
23/23 - 0s - loss: 0.4685 - 16ms/epoch - 674us/step
Epoch 41/100
23/23 - 0s - loss: 0.7208 - 16ms/epoch - 679us/step
Epoch 42/100
23/23 - 0s - loss: 0.7944 - 15ms/epoch - 664us/step
Epoch 43/100
23/23 - 0s - loss: 0.6543 - 16ms/epoch - 676us/step
Epoch 44/100
23/23 - 0s - loss: 0.6341 - 16ms/epoch - 680us/step
Epoch 45/100
23/23 - 0s - loss: 0.4636 - 16ms/epoch - 677us/step
Epoch 46/100
23/23 - 0s - loss: 0.4347 - 16ms/epoch - 686us/step
Epoch 47/100
23/23 - 0s - loss: 0.5630 - 16ms/epoch - 711us/step
Epoch 48/100
23/23 - 0s - loss: 0.5360 - 15ms/epoch - 665us/step
Epoch 49/100
23/23 - 0s - loss: 0.7723 - 18ms/epoch - 766us/step
Epoch 50/100
23/23 - 0s - loss: 0.5627 - 16ms/epoch - 684us/step
Epoch 51/100
23/23 - 0s - loss: 0.5036 - 15ms/epoch - 671us/step
Epoch 52/100
23/23 - 0s - loss: 0.6356 - 16ms/epoch - 678us/step
Epoch 53/100
23/23 - 0s - loss: 0.4985 - 15ms/epoch - 672us/step
Epoch 54/100
23/23 - 0s - loss: 0.5062 - 16ms/epoch - 679us/step
Epoch 55/100
23/23 - 0s -

Epoch 66/100
23/23 - 0s - loss: 0.4930 - 16ms/epoch - 678us/step
Epoch 67/100
23/23 - 0s - loss: 0.7145 - 16ms/epoch - 686us/step
Epoch 68/100
23/23 - 0s - loss: 0.4613 - 16ms/epoch - 690us/step
Epoch 69/100
23/23 - 0s - loss: 0.6099 - 16ms/epoch - 688us/step
Epoch 70/100
23/23 - 0s - loss: 0.6806 - 16ms/epoch - 694us/step
Epoch 71/100
23/23 - 0s - loss: 0.6130 - 16ms/epoch - 707us/step
Epoch 72/100
23/23 - 0s - loss: 0.7810 - 16ms/epoch - 697us/step
Epoch 73/100
23/23 - 0s - loss: 0.5474 - 16ms/epoch - 690us/step
Epoch 74/100
23/23 - 0s - loss: 0.5221 - 16ms/epoch - 694us/step
Epoch 75/100
23/23 - 0s - loss: 0.4873 - 16ms/epoch - 694us/step
Epoch 76/100
23/23 - 0s - loss: 0.4907 - 16ms/epoch - 691us/step
Epoch 77/100
23/23 - 0s - loss: 0.6837 - 17ms/epoch - 729us/step
Epoch 78/100
23/23 - 0s - loss: 0.5263 - 17ms/epoch - 726us/step
Epoch 79/100
23/23 - 0s - loss: 0.6095 - 17ms/epoch - 745us/step
Epoch 80/100
23/23 - 0s - loss: 0.4905 - 17ms/epoch - 732us/step
Epoch 81/100
23/23 - 0s -

Epoch 92/100
23/23 - 0s - loss: 0.5251 - 22ms/epoch - 937us/step
Epoch 93/100
23/23 - 0s - loss: 0.5023 - 20ms/epoch - 883us/step
Epoch 94/100
23/23 - 0s - loss: 0.5104 - 19ms/epoch - 808us/step
Epoch 95/100
23/23 - 0s - loss: 0.5063 - 19ms/epoch - 832us/step
Epoch 96/100
23/23 - 0s - loss: 0.4205 - 20ms/epoch - 880us/step
Epoch 97/100
23/23 - 0s - loss: 0.5865 - 18ms/epoch - 774us/step
Epoch 98/100
23/23 - 0s - loss: 0.5030 - 18ms/epoch - 796us/step
Epoch 99/100
23/23 - 0s - loss: 0.5046 - 18ms/epoch - 767us/step
Epoch 100/100
23/23 - 0s - loss: 0.5972 - 18ms/epoch - 787us/step
10/10 [==============================] - 0s 916us/step
Epoch 1/100
23/23 - 0s - loss: 0.5272 - 20ms/epoch - 881us/step
Epoch 2/100
23/23 - 0s - loss: 0.5561 - 19ms/epoch - 806us/step
Epoch 3/100
23/23 - 0s - loss: 0.7761 - 18ms/epoch - 775us/step
Epoch 4/100
23/23 - 0s - loss: 0.4966 - 16ms/epoch - 698us/step
Epoch 5/100
23/23 - 0s - loss: 0.4921 - 16ms/epoch - 678us/step
Epoch 6/100
23/23 - 0s - loss: 0.4264 -

Epoch 17/100
23/23 - 0s - loss: 0.5641 - 16ms/epoch - 687us/step
Epoch 18/100
23/23 - 0s - loss: 0.5894 - 15ms/epoch - 673us/step
Epoch 19/100
23/23 - 0s - loss: 0.6310 - 24ms/epoch - 1ms/step
Epoch 20/100
23/23 - 0s - loss: 0.5627 - 17ms/epoch - 725us/step
Epoch 21/100
23/23 - 0s - loss: 0.5984 - 16ms/epoch - 696us/step
Epoch 22/100
23/23 - 0s - loss: 0.5828 - 15ms/epoch - 666us/step
Epoch 23/100
23/23 - 0s - loss: 0.6008 - 16ms/epoch - 699us/step
Epoch 24/100
23/23 - 0s - loss: 0.6442 - 17ms/epoch - 742us/step
Epoch 25/100
23/23 - 0s - loss: 0.7052 - 16ms/epoch - 677us/step
Epoch 26/100
23/23 - 0s - loss: 0.5607 - 16ms/epoch - 687us/step
Epoch 27/100
23/23 - 0s - loss: 0.4347 - 16ms/epoch - 678us/step
Epoch 28/100
23/23 - 0s - loss: 0.4931 - 22ms/epoch - 966us/step
Epoch 29/100
23/23 - 0s - loss: 0.4430 - 18ms/epoch - 783us/step
Epoch 30/100
23/23 - 0s - loss: 0.4489 - 17ms/epoch - 721us/step
Epoch 31/100
23/23 - 0s - loss: 0.5577 - 16ms/epoch - 686us/step
Epoch 32/100
23/23 - 0s - l

Epoch 43/100
23/23 - 0s - loss: 0.4532 - 15ms/epoch - 666us/step
Epoch 44/100
23/23 - 0s - loss: 0.5056 - 16ms/epoch - 698us/step
Epoch 45/100
23/23 - 0s - loss: 0.6569 - 15ms/epoch - 673us/step
Epoch 46/100
23/23 - 0s - loss: 0.4832 - 15ms/epoch - 669us/step
Epoch 47/100
23/23 - 0s - loss: 0.4256 - 15ms/epoch - 672us/step
Epoch 48/100
23/23 - 0s - loss: 0.4307 - 16ms/epoch - 675us/step
Epoch 49/100
23/23 - 0s - loss: 0.4325 - 16ms/epoch - 689us/step
Epoch 50/100
23/23 - 0s - loss: 0.6204 - 18ms/epoch - 801us/step
Epoch 51/100
23/23 - 0s - loss: 0.5056 - 16ms/epoch - 707us/step
Epoch 52/100
23/23 - 0s - loss: 0.4578 - 21ms/epoch - 926us/step
Epoch 53/100
23/23 - 0s - loss: 0.5781 - 18ms/epoch - 789us/step
Epoch 54/100
23/23 - 0s - loss: 0.4564 - 17ms/epoch - 748us/step
Epoch 55/100
23/23 - 0s - loss: 0.4340 - 16ms/epoch - 676us/step
Epoch 56/100
23/23 - 0s - loss: 0.4527 - 16ms/epoch - 678us/step
Epoch 57/100
23/23 - 0s - loss: 0.5199 - 17ms/epoch - 729us/step
Epoch 58/100
23/23 - 0s -

Epoch 69/100
23/23 - 0s - loss: 0.4284 - 16ms/epoch - 684us/step
Epoch 70/100
23/23 - 0s - loss: 0.4787 - 22ms/epoch - 965us/step
Epoch 71/100
23/23 - 0s - loss: 0.6403 - 18ms/epoch - 778us/step
Epoch 72/100
23/23 - 0s - loss: 0.5489 - 16ms/epoch - 693us/step
Epoch 73/100
23/23 - 0s - loss: 0.7184 - 17ms/epoch - 725us/step
Epoch 74/100
23/23 - 0s - loss: 0.5609 - 25ms/epoch - 1ms/step
Epoch 75/100
23/23 - 0s - loss: 0.5037 - 20ms/epoch - 855us/step
Epoch 76/100
23/23 - 0s - loss: 0.7124 - 16ms/epoch - 698us/step
Epoch 77/100
23/23 - 0s - loss: 0.5918 - 17ms/epoch - 752us/step
Epoch 78/100
23/23 - 0s - loss: 0.5006 - 17ms/epoch - 727us/step
Epoch 79/100
23/23 - 0s - loss: 0.4452 - 16ms/epoch - 694us/step
Epoch 80/100
23/23 - 0s - loss: 0.7498 - 16ms/epoch - 695us/step
Epoch 81/100
23/23 - 0s - loss: 0.4688 - 16ms/epoch - 709us/step
Epoch 82/100
23/23 - 0s - loss: 0.6870 - 16ms/epoch - 683us/step
Epoch 83/100
23/23 - 0s - loss: 0.5203 - 16ms/epoch - 688us/step
Epoch 84/100
23/23 - 0s - l

Epoch 95/100
23/23 - 0s - loss: 0.6012 - 16ms/epoch - 678us/step
Epoch 96/100
23/23 - 0s - loss: 0.5247 - 16ms/epoch - 681us/step
Epoch 97/100
23/23 - 0s - loss: 0.4802 - 17ms/epoch - 722us/step
Epoch 98/100
23/23 - 0s - loss: 0.4820 - 15ms/epoch - 673us/step
Epoch 99/100
23/23 - 0s - loss: 0.4818 - 16ms/epoch - 686us/step
Epoch 100/100
23/23 - 0s - loss: 0.4788 - 15ms/epoch - 666us/step
10/10 [==============================] - 0s 1ms/step
Epoch 1/100
23/23 - 0s - loss: 0.5179 - 19ms/epoch - 822us/step
Epoch 2/100
23/23 - 0s - loss: 0.5938 - 16ms/epoch - 697us/step
Epoch 3/100
23/23 - 0s - loss: 0.4864 - 16ms/epoch - 696us/step
Epoch 4/100
23/23 - 0s - loss: 0.4566 - 16ms/epoch - 693us/step
Epoch 5/100
23/23 - 0s - loss: 0.5113 - 16ms/epoch - 686us/step
Epoch 6/100
23/23 - 0s - loss: 0.4367 - 17ms/epoch - 720us/step
Epoch 7/100
23/23 - 0s - loss: 0.4604 - 15ms/epoch - 671us/step
Epoch 8/100
23/23 - 0s - loss: 0.4564 - 16ms/epoch - 676us/step
Epoch 9/100
23/23 - 0s - loss: 0.4477 - 15ms

Epoch 20/100
23/23 - 0s - loss: 0.4511 - 17ms/epoch - 730us/step
Epoch 21/100
23/23 - 0s - loss: 0.4735 - 17ms/epoch - 722us/step
Epoch 22/100
23/23 - 0s - loss: 0.5406 - 17ms/epoch - 760us/step
Epoch 23/100
23/23 - 0s - loss: 0.4509 - 17ms/epoch - 754us/step
Epoch 24/100
23/23 - 0s - loss: 0.4722 - 17ms/epoch - 738us/step
Epoch 25/100
23/23 - 0s - loss: 0.4794 - 18ms/epoch - 797us/step
Epoch 26/100
23/23 - 0s - loss: 0.5638 - 18ms/epoch - 766us/step
Epoch 27/100
23/23 - 0s - loss: 0.4903 - 18ms/epoch - 779us/step
Epoch 28/100
23/23 - 0s - loss: 0.4645 - 17ms/epoch - 755us/step
Epoch 29/100
23/23 - 0s - loss: 0.5094 - 17ms/epoch - 741us/step
Epoch 30/100
23/23 - 0s - loss: 0.5890 - 17ms/epoch - 722us/step
Epoch 31/100
23/23 - 0s - loss: 0.5829 - 17ms/epoch - 737us/step
Epoch 32/100
23/23 - 0s - loss: 0.5205 - 17ms/epoch - 727us/step
Epoch 33/100
23/23 - 0s - loss: 0.5355 - 17ms/epoch - 734us/step
Epoch 34/100
23/23 - 0s - loss: 0.5991 - 17ms/epoch - 729us/step
Epoch 35/100
23/23 - 0s -

Epoch 46/100
23/23 - 0s - loss: 0.5291 - 16ms/epoch - 682us/step
Epoch 47/100
23/23 - 0s - loss: 1.0143 - 16ms/epoch - 675us/step
Epoch 48/100
23/23 - 0s - loss: 0.5717 - 15ms/epoch - 663us/step
Epoch 49/100
23/23 - 0s - loss: 0.4449 - 15ms/epoch - 670us/step
Epoch 50/100
23/23 - 0s - loss: 0.7497 - 16ms/epoch - 678us/step
Epoch 51/100
23/23 - 0s - loss: 0.5463 - 16ms/epoch - 681us/step
Epoch 52/100
23/23 - 0s - loss: 0.6010 - 16ms/epoch - 676us/step
Epoch 53/100
23/23 - 0s - loss: 0.4586 - 16ms/epoch - 682us/step
Epoch 54/100
23/23 - 0s - loss: 0.6242 - 16ms/epoch - 694us/step
Epoch 55/100
23/23 - 0s - loss: 0.8000 - 16ms/epoch - 690us/step
Epoch 56/100
23/23 - 0s - loss: 0.5096 - 17ms/epoch - 742us/step
Epoch 57/100
23/23 - 0s - loss: 0.5845 - 18ms/epoch - 799us/step
Epoch 58/100
23/23 - 0s - loss: 0.4956 - 17ms/epoch - 747us/step
Epoch 59/100
23/23 - 0s - loss: 0.7052 - 17ms/epoch - 733us/step
Epoch 60/100
23/23 - 0s - loss: 0.5619 - 17ms/epoch - 749us/step
Epoch 61/100
23/23 - 0s -

Epoch 72/100
23/23 - 0s - loss: 0.4592 - 17ms/epoch - 746us/step
Epoch 73/100
23/23 - 0s - loss: 0.4338 - 17ms/epoch - 731us/step
Epoch 74/100
23/23 - 0s - loss: 0.4977 - 17ms/epoch - 723us/step
Epoch 75/100
23/23 - 0s - loss: 0.4489 - 17ms/epoch - 724us/step
Epoch 76/100
23/23 - 0s - loss: 0.4504 - 17ms/epoch - 748us/step
Epoch 77/100
23/23 - 0s - loss: 0.6035 - 17ms/epoch - 719us/step
Epoch 78/100
23/23 - 0s - loss: 0.6728 - 17ms/epoch - 756us/step
Epoch 79/100
23/23 - 0s - loss: 0.5540 - 19ms/epoch - 807us/step
Epoch 80/100
23/23 - 0s - loss: 0.5033 - 17ms/epoch - 741us/step
Epoch 81/100
23/23 - 0s - loss: 0.4753 - 18ms/epoch - 782us/step
Epoch 82/100
23/23 - 0s - loss: 0.4921 - 18ms/epoch - 778us/step
Epoch 83/100
23/23 - 0s - loss: 0.5666 - 16ms/epoch - 714us/step
Epoch 84/100
23/23 - 0s - loss: 0.5567 - 16ms/epoch - 686us/step
Epoch 85/100
23/23 - 0s - loss: 0.5814 - 24ms/epoch - 1ms/step
Epoch 86/100
23/23 - 0s - loss: 0.8192 - 20ms/epoch - 863us/step
Epoch 87/100
23/23 - 0s - l

Epoch 98/100
23/23 - 0s - loss: 0.4840 - 16ms/epoch - 705us/step
Epoch 99/100
23/23 - 0s - loss: 0.5273 - 16ms/epoch - 678us/step
Epoch 100/100
23/23 - 0s - loss: 0.4473 - 16ms/epoch - 688us/step
10/10 [==============================] - 0s 890us/step
Epoch 1/100
23/23 - 0s - loss: 0.5326 - 19ms/epoch - 818us/step
Epoch 2/100
23/23 - 0s - loss: 0.6050 - 17ms/epoch - 722us/step
Epoch 3/100
23/23 - 0s - loss: 0.4539 - 17ms/epoch - 735us/step
Epoch 4/100
23/23 - 0s - loss: 0.4471 - 18ms/epoch - 799us/step
Epoch 5/100
23/23 - 0s - loss: 0.4413 - 17ms/epoch - 751us/step
Epoch 6/100
23/23 - 0s - loss: 0.6608 - 17ms/epoch - 758us/step
Epoch 7/100
23/23 - 0s - loss: 0.6388 - 17ms/epoch - 732us/step
Epoch 8/100
23/23 - 0s - loss: 0.4620 - 17ms/epoch - 741us/step
Epoch 9/100
23/23 - 0s - loss: 0.5087 - 17ms/epoch - 727us/step
Epoch 10/100
23/23 - 0s - loss: 0.5372 - 16ms/epoch - 709us/step
Epoch 11/100
23/23 - 0s - loss: 0.5736 - 20ms/epoch - 862us/step
Epoch 12/100
23/23 - 0s - loss: 0.5588 - 16

Epoch 23/100
23/23 - 0s - loss: 0.5763 - 16ms/epoch - 693us/step
Epoch 24/100
23/23 - 0s - loss: 0.8158 - 16ms/epoch - 678us/step
Epoch 25/100
23/23 - 0s - loss: 0.4927 - 16ms/epoch - 699us/step
Epoch 26/100
23/23 - 0s - loss: 0.4312 - 16ms/epoch - 683us/step
Epoch 27/100
23/23 - 0s - loss: 0.4774 - 16ms/epoch - 680us/step
Epoch 28/100
23/23 - 0s - loss: 0.5184 - 16ms/epoch - 713us/step
Epoch 29/100
23/23 - 0s - loss: 0.4312 - 17ms/epoch - 732us/step
Epoch 30/100
23/23 - 0s - loss: 0.6966 - 16ms/epoch - 707us/step
Epoch 31/100
23/23 - 0s - loss: 0.4914 - 16ms/epoch - 694us/step
Epoch 32/100
23/23 - 0s - loss: 0.4460 - 18ms/epoch - 772us/step
Epoch 33/100
23/23 - 0s - loss: 0.4501 - 17ms/epoch - 746us/step
Epoch 34/100
23/23 - 0s - loss: 0.6945 - 17ms/epoch - 733us/step
Epoch 35/100
23/23 - 0s - loss: 0.6292 - 17ms/epoch - 738us/step
Epoch 36/100
23/23 - 0s - loss: 0.4762 - 17ms/epoch - 754us/step
Epoch 37/100
23/23 - 0s - loss: 0.4770 - 17ms/epoch - 752us/step
Epoch 38/100
23/23 - 0s -

Epoch 49/100
23/23 - 0s - loss: 0.5715 - 16ms/epoch - 709us/step
Epoch 50/100
23/23 - 0s - loss: 0.6362 - 16ms/epoch - 699us/step
Epoch 51/100
23/23 - 0s - loss: 0.6665 - 16ms/epoch - 705us/step
Epoch 52/100
23/23 - 0s - loss: 0.5130 - 17ms/epoch - 725us/step
Epoch 53/100
23/23 - 0s - loss: 0.5915 - 17ms/epoch - 749us/step
Epoch 54/100
23/23 - 0s - loss: 0.5852 - 17ms/epoch - 739us/step
Epoch 55/100
23/23 - 0s - loss: 0.4941 - 17ms/epoch - 760us/step
Epoch 56/100
23/23 - 0s - loss: 0.4900 - 17ms/epoch - 735us/step
Epoch 57/100
23/23 - 0s - loss: 0.6542 - 17ms/epoch - 746us/step
Epoch 58/100
23/23 - 0s - loss: 0.4582 - 18ms/epoch - 769us/step
Epoch 59/100
23/23 - 0s - loss: 0.5186 - 21ms/epoch - 901us/step
Epoch 60/100
23/23 - 0s - loss: 0.4461 - 18ms/epoch - 776us/step
Epoch 61/100
23/23 - 0s - loss: 0.4801 - 17ms/epoch - 738us/step
Epoch 62/100
23/23 - 0s - loss: 0.5295 - 18ms/epoch - 774us/step
Epoch 63/100
23/23 - 0s - loss: 0.5215 - 18ms/epoch - 784us/step
Epoch 64/100
23/23 - 0s -

Epoch 75/100
23/23 - 0s - loss: 0.5519 - 17ms/epoch - 745us/step
Epoch 76/100
23/23 - 0s - loss: 0.5100 - 16ms/epoch - 680us/step
Epoch 77/100
23/23 - 0s - loss: 0.6145 - 16ms/epoch - 684us/step
Epoch 78/100
23/23 - 0s - loss: 0.5609 - 16ms/epoch - 693us/step
Epoch 79/100
23/23 - 0s - loss: 0.4432 - 16ms/epoch - 681us/step
Epoch 80/100
23/23 - 0s - loss: 0.5759 - 22ms/epoch - 949us/step
Epoch 81/100
23/23 - 0s - loss: 0.4292 - 17ms/epoch - 755us/step
Epoch 82/100
23/23 - 0s - loss: 0.4757 - 16ms/epoch - 702us/step
Epoch 83/100
23/23 - 0s - loss: 0.8161 - 16ms/epoch - 684us/step
Epoch 84/100
23/23 - 0s - loss: 0.7110 - 15ms/epoch - 669us/step
Epoch 85/100
23/23 - 0s - loss: 0.4614 - 15ms/epoch - 672us/step
Epoch 86/100
23/23 - 0s - loss: 0.9257 - 16ms/epoch - 678us/step
Epoch 87/100
23/23 - 0s - loss: 0.6158 - 16ms/epoch - 681us/step
Epoch 88/100
23/23 - 0s - loss: 0.4653 - 16ms/epoch - 678us/step
Epoch 89/100
23/23 - 0s - loss: 0.4859 - 16ms/epoch - 688us/step
Epoch 90/100
23/23 - 0s -

10/10 [==============================] - 0s 833us/step
Epoch 1/100
23/23 - 0s - loss: 0.5689 - 19ms/epoch - 844us/step
Epoch 2/100
23/23 - 0s - loss: 0.9802 - 16ms/epoch - 696us/step
Epoch 3/100
23/23 - 0s - loss: 0.8786 - 16ms/epoch - 681us/step
Epoch 4/100
23/23 - 0s - loss: 0.6586 - 20ms/epoch - 857us/step
Epoch 5/100
23/23 - 0s - loss: 0.5933 - 18ms/epoch - 764us/step
Epoch 6/100
23/23 - 0s - loss: 0.4536 - 16ms/epoch - 686us/step
Epoch 7/100
23/23 - 0s - loss: 0.4460 - 20ms/epoch - 884us/step
Epoch 8/100
23/23 - 0s - loss: 0.6349 - 18ms/epoch - 785us/step
Epoch 9/100
23/23 - 0s - loss: 0.4602 - 17ms/epoch - 736us/step
Epoch 10/100
23/23 - 0s - loss: 0.5085 - 15ms/epoch - 673us/step
Epoch 11/100
23/23 - 0s - loss: 0.5146 - 17ms/epoch - 722us/step
Epoch 12/100
23/23 - 0s - loss: 0.4936 - 16ms/epoch - 679us/step
Epoch 13/100
23/23 - 0s - loss: 0.4244 - 16ms/epoch - 679us/step
Epoch 14/100
23/23 - 0s - loss: 0.6067 - 16ms/epoch - 675us/step
Epoch 15/100
23/23 - 0s - loss: 0.5403 - 16m

Epoch 26/100
23/23 - 0s - loss: 0.5340 - 18ms/epoch - 782us/step
Epoch 27/100
23/23 - 0s - loss: 0.5639 - 15ms/epoch - 669us/step
Epoch 28/100
23/23 - 0s - loss: 0.5497 - 22ms/epoch - 978us/step
Epoch 29/100
23/23 - 0s - loss: 0.6375 - 17ms/epoch - 720us/step
Epoch 30/100
23/23 - 0s - loss: 0.5100 - 16ms/epoch - 712us/step
Epoch 31/100
23/23 - 0s - loss: 0.5300 - 15ms/epoch - 663us/step
Epoch 32/100
23/23 - 0s - loss: 0.5460 - 15ms/epoch - 664us/step
Epoch 33/100
23/23 - 0s - loss: 0.7473 - 15ms/epoch - 664us/step
Epoch 34/100
23/23 - 0s - loss: 0.7738 - 15ms/epoch - 663us/step
Epoch 35/100
23/23 - 0s - loss: 0.4679 - 16ms/epoch - 706us/step
Epoch 36/100
23/23 - 0s - loss: 0.4538 - 20ms/epoch - 866us/step
Epoch 37/100
23/23 - 0s - loss: 0.4206 - 16ms/epoch - 682us/step
Epoch 38/100
23/23 - 0s - loss: 0.5201 - 15ms/epoch - 673us/step
Epoch 39/100
23/23 - 0s - loss: 0.5559 - 16ms/epoch - 694us/step
Epoch 40/100
23/23 - 0s - loss: 0.5421 - 16ms/epoch - 702us/step
Epoch 41/100
23/23 - 0s -

Epoch 52/100
23/23 - 0s - loss: 0.4443 - 15ms/epoch - 668us/step
Epoch 53/100
23/23 - 0s - loss: 0.5099 - 20ms/epoch - 887us/step
Epoch 54/100
23/23 - 0s - loss: 0.5209 - 16ms/epoch - 699us/step
Epoch 55/100
23/23 - 0s - loss: 0.7217 - 16ms/epoch - 677us/step
Epoch 56/100
23/23 - 0s - loss: 0.5567 - 16ms/epoch - 692us/step
Epoch 57/100
23/23 - 0s - loss: 0.4988 - 15ms/epoch - 662us/step
Epoch 58/100
23/23 - 0s - loss: 0.5141 - 17ms/epoch - 718us/step
Epoch 59/100
23/23 - 0s - loss: 0.4990 - 16ms/epoch - 675us/step
Epoch 60/100
23/23 - 0s - loss: 0.4408 - 15ms/epoch - 673us/step
Epoch 61/100
23/23 - 0s - loss: 0.4956 - 16ms/epoch - 676us/step
Epoch 62/100
23/23 - 0s - loss: 0.5098 - 16ms/epoch - 678us/step
Epoch 63/100
23/23 - 0s - loss: 0.4393 - 15ms/epoch - 664us/step
Epoch 64/100
23/23 - 0s - loss: 0.4585 - 18ms/epoch - 765us/step
Epoch 65/100
23/23 - 0s - loss: 0.6638 - 17ms/epoch - 745us/step
Epoch 66/100
23/23 - 0s - loss: 0.6214 - 16ms/epoch - 684us/step
Epoch 67/100
23/23 - 0s -

Epoch 78/100
23/23 - 0s - loss: 0.7422 - 19ms/epoch - 822us/step
Epoch 79/100
23/23 - 0s - loss: 0.5857 - 23ms/epoch - 1ms/step
Epoch 80/100
23/23 - 0s - loss: 0.4726 - 23ms/epoch - 981us/step
Epoch 81/100
23/23 - 0s - loss: 0.4425 - 19ms/epoch - 833us/step
Epoch 82/100
23/23 - 0s - loss: 0.4486 - 22ms/epoch - 977us/step
Epoch 83/100
23/23 - 0s - loss: 0.4209 - 20ms/epoch - 861us/step
Epoch 84/100
23/23 - 0s - loss: 0.5702 - 20ms/epoch - 888us/step
Epoch 85/100
23/23 - 0s - loss: 0.5186 - 19ms/epoch - 842us/step
Epoch 86/100
23/23 - 0s - loss: 0.6595 - 18ms/epoch - 793us/step
Epoch 87/100
23/23 - 0s - loss: 0.4521 - 17ms/epoch - 754us/step
Epoch 88/100
23/23 - 0s - loss: 0.4486 - 19ms/epoch - 835us/step
Epoch 89/100
23/23 - 0s - loss: 0.4458 - 23ms/epoch - 987us/step
Epoch 90/100
23/23 - 0s - loss: 0.4503 - 22ms/epoch - 944us/step
Epoch 91/100
23/23 - 0s - loss: 0.5158 - 20ms/epoch - 871us/step
Epoch 92/100
23/23 - 0s - loss: 0.5663 - 18ms/epoch - 777us/step
Epoch 93/100
23/23 - 0s - l

Epoch 3/100
23/23 - 0s - loss: 0.5131 - 15ms/epoch - 674us/step
Epoch 4/100
23/23 - 0s - loss: 0.4898 - 16ms/epoch - 693us/step
Epoch 5/100
23/23 - 0s - loss: 0.4743 - 16ms/epoch - 689us/step
Epoch 6/100
23/23 - 0s - loss: 0.6055 - 16ms/epoch - 687us/step
Epoch 7/100
23/23 - 0s - loss: 0.4653 - 20ms/epoch - 853us/step
Epoch 8/100
23/23 - 0s - loss: 0.6034 - 18ms/epoch - 773us/step
Epoch 9/100
23/23 - 0s - loss: 0.6030 - 16ms/epoch - 676us/step
Epoch 10/100
23/23 - 0s - loss: 0.5842 - 17ms/epoch - 737us/step
Epoch 11/100
23/23 - 0s - loss: 0.6654 - 17ms/epoch - 742us/step
Epoch 12/100
23/23 - 0s - loss: 0.4662 - 16ms/epoch - 689us/step
Epoch 13/100
23/23 - 0s - loss: 0.4724 - 16ms/epoch - 688us/step
Epoch 14/100
23/23 - 0s - loss: 0.4435 - 16ms/epoch - 689us/step
Epoch 15/100
23/23 - 0s - loss: 0.4369 - 17ms/epoch - 752us/step
Epoch 16/100
23/23 - 0s - loss: 0.4400 - 18ms/epoch - 790us/step
Epoch 17/100
23/23 - 0s - loss: 0.5132 - 16ms/epoch - 680us/step
Epoch 18/100
23/23 - 0s - loss: 

Epoch 29/100
23/23 - 0s - loss: 0.5297 - 16ms/epoch - 681us/step
Epoch 30/100
23/23 - 0s - loss: 0.5151 - 16ms/epoch - 691us/step
Epoch 31/100
23/23 - 0s - loss: 0.5515 - 16ms/epoch - 691us/step
Epoch 32/100
23/23 - 0s - loss: 0.5516 - 16ms/epoch - 697us/step
Epoch 33/100
23/23 - 0s - loss: 0.6208 - 16ms/epoch - 677us/step
Epoch 34/100
23/23 - 0s - loss: 0.5473 - 15ms/epoch - 669us/step
Epoch 35/100
23/23 - 0s - loss: 0.5603 - 16ms/epoch - 683us/step
Epoch 36/100
23/23 - 0s - loss: 0.5851 - 16ms/epoch - 710us/step
Epoch 37/100
23/23 - 0s - loss: 0.4861 - 18ms/epoch - 763us/step
Epoch 38/100
23/23 - 0s - loss: 0.4275 - 18ms/epoch - 766us/step
Epoch 39/100
23/23 - 0s - loss: 0.4742 - 17ms/epoch - 742us/step
Epoch 40/100
23/23 - 0s - loss: 0.5208 - 17ms/epoch - 743us/step
Epoch 41/100
23/23 - 0s - loss: 0.4886 - 17ms/epoch - 731us/step
Epoch 42/100
23/23 - 0s - loss: 0.6539 - 17ms/epoch - 747us/step
Epoch 43/100
23/23 - 0s - loss: 0.5680 - 17ms/epoch - 745us/step
Epoch 44/100
23/23 - 0s -

Epoch 55/100
23/23 - 0s - loss: 0.8693 - 16ms/epoch - 691us/step
Epoch 56/100
23/23 - 0s - loss: 0.5066 - 16ms/epoch - 707us/step
Epoch 57/100
23/23 - 0s - loss: 0.4388 - 16ms/epoch - 708us/step
Epoch 58/100
23/23 - 0s - loss: 0.4844 - 18ms/epoch - 790us/step
Epoch 59/100
23/23 - 0s - loss: 0.4743 - 18ms/epoch - 776us/step
Epoch 60/100
23/23 - 0s - loss: 0.7915 - 18ms/epoch - 765us/step
Epoch 61/100
23/23 - 0s - loss: 0.8077 - 17ms/epoch - 745us/step
Epoch 62/100
23/23 - 0s - loss: 0.5677 - 17ms/epoch - 733us/step
Epoch 63/100
23/23 - 0s - loss: 0.5063 - 17ms/epoch - 723us/step
Epoch 64/100
23/23 - 0s - loss: 0.4935 - 17ms/epoch - 724us/step
Epoch 65/100
23/23 - 0s - loss: 0.7051 - 17ms/epoch - 729us/step
Epoch 66/100
23/23 - 0s - loss: 0.5480 - 17ms/epoch - 731us/step
Epoch 67/100
23/23 - 0s - loss: 0.4430 - 17ms/epoch - 734us/step
Epoch 68/100
23/23 - 0s - loss: 0.5013 - 16ms/epoch - 679us/step
Epoch 69/100
23/23 - 0s - loss: 0.4716 - 16ms/epoch - 703us/step
Epoch 70/100
23/23 - 0s -

Epoch 81/100
23/23 - 0s - loss: 0.4579 - 17ms/epoch - 728us/step
Epoch 82/100
23/23 - 0s - loss: 0.4181 - 17ms/epoch - 744us/step
Epoch 83/100
23/23 - 0s - loss: 0.5075 - 17ms/epoch - 725us/step
Epoch 84/100
23/23 - 0s - loss: 0.4807 - 17ms/epoch - 744us/step
Epoch 85/100
23/23 - 0s - loss: 0.5127 - 16ms/epoch - 715us/step
Epoch 86/100
23/23 - 0s - loss: 0.6191 - 17ms/epoch - 732us/step
Epoch 87/100
23/23 - 0s - loss: 0.7228 - 16ms/epoch - 692us/step
Epoch 88/100
23/23 - 0s - loss: 0.5308 - 17ms/epoch - 733us/step
Epoch 89/100
23/23 - 0s - loss: 0.5237 - 18ms/epoch - 763us/step
Epoch 90/100
23/23 - 0s - loss: 0.4330 - 17ms/epoch - 752us/step
Epoch 91/100
23/23 - 0s - loss: 0.6538 - 17ms/epoch - 729us/step
Epoch 92/100
23/23 - 0s - loss: 0.5311 - 18ms/epoch - 784us/step
Epoch 93/100
23/23 - 0s - loss: 0.5263 - 17ms/epoch - 742us/step
Epoch 94/100
23/23 - 0s - loss: 0.4712 - 17ms/epoch - 727us/step
Epoch 95/100
23/23 - 0s - loss: 0.4691 - 17ms/epoch - 741us/step
Epoch 96/100
23/23 - 0s -

Epoch 6/100
23/23 - 0s - loss: 0.5883 - 16ms/epoch - 696us/step
Epoch 7/100
23/23 - 0s - loss: 0.4882 - 16ms/epoch - 690us/step
Epoch 8/100
23/23 - 0s - loss: 0.4666 - 16ms/epoch - 711us/step
Epoch 9/100
23/23 - 0s - loss: 0.5579 - 16ms/epoch - 705us/step
Epoch 10/100
23/23 - 0s - loss: 0.6336 - 16ms/epoch - 684us/step
Epoch 11/100
23/23 - 0s - loss: 0.5852 - 16ms/epoch - 690us/step
Epoch 12/100
23/23 - 0s - loss: 0.4440 - 16ms/epoch - 704us/step
Epoch 13/100
23/23 - 0s - loss: 0.5333 - 18ms/epoch - 762us/step
Epoch 14/100
23/23 - 0s - loss: 0.5990 - 17ms/epoch - 722us/step
Epoch 15/100
23/23 - 0s - loss: 0.4938 - 19ms/epoch - 835us/step
Epoch 16/100
23/23 - 0s - loss: 0.4650 - 18ms/epoch - 791us/step
Epoch 17/100
23/23 - 0s - loss: 0.4374 - 60ms/epoch - 3ms/step
Epoch 18/100
23/23 - 0s - loss: 0.4202 - 17ms/epoch - 723us/step
Epoch 19/100
23/23 - 0s - loss: 0.4540 - 16ms/epoch - 699us/step
Epoch 20/100
23/23 - 0s - loss: 0.4563 - 16ms/epoch - 680us/step
Epoch 21/100
23/23 - 0s - loss:

### Comparison to the mean of the MSEs compred to which from A

In [5]:
print("The final MSE at the 50th time is {}.".format(MSE_list[-1]))
print("The mean of the MSE is {}.".format(np.mean(MSE_list)))
print("The standard deviation of the MSE is {}".format(np.std(MSE_list)))


The final MSE at the 50th time is 0.6785354998283251.
The mean of the MSE is 0.5262634357577329.
The standard deviation of the MSE is 0.1671090582276595


## How does the mean of the MSEs compare to which from Step B?

#### Mean of MSEs from Step B: 0.7051157050923686
#### Mean of MSEs from Step C: 0.5262634357577329
### Thus, the mean of the MSEs is decreased compare to which from Step B.